src/main/java/org/codehaus/plexus/archiver/AbstractUnArchiver.java

```diff
@@ -384,4 +384,39 @@ else if ( isDirectory )
         }
     }
 
+    // Visible for testing
+    protected boolean shouldExtractEntry( File file, String entryName, Date entryDate ) throws IOException
+    {
+        String canonicalDestPath = file.getCanonicalPath();
```

Comment:
This may be expensive to calculate. I had JIRA issues because of this for Maven Resolver. Do you really need the canonical path?

Reply:
Well, it was already there before I picked up this issue.

Didn't know it was expensive. But now we're doing it twice, basically to keep the code short & readable. There is caching for the result (in the `java.io.FileSystem` class), enabled by default, so I am tempted to leave it here unless you have a strong opinion about it. We could pass it in from the calling method alternatively.

Reply:
OK, leave it as-is.